runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,167 @@
+import javax.inject.Inject;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.api.Client;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.events.NpcDespawned;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.api.GameState;
+import net.runelite.api.NPC;
+import net.runelite.api.NpcID;
+import net.runelite.api.Player;
+import net.runelite.api.Varbits;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Stealing Artefacts",
+	description = "Show the location and status of your Stealing Artefact task.",
+	tags = {"thieving", "stealing", "artefact", "artefacts", "captain", "khaled"},
+	enabledByDefault = false
+)
+
+public class StealingArtefactsPlugin extends Plugin
+{
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean inPortPiscariliusRegion;
+
+	private NPC captainKhaled;
+
+	@Override
+	protected void startUp()
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.values()[client.getVar(Varbits.STEALING_ARTEFACT_STATE)]);
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		client.clearHintArrow();
+		stealingArtefactState = null;
+		captainKhaled = null;
+		overlayManager.remove(overlay);
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		StealingArtefactState state = StealingArtefactState.values()[(client.getVar(Varbits.STEALING_ARTEFACT_STATE))];
+		stealingArtefactState = state;
+
+		if (state == StealingArtefactState.DELIVER_ARTEFACT && captainKhaled != null)
+		{
+			client.setHintArrow(captainKhaled);
+		}
+		else if (state == StealingArtefactState.NO_TASK || state == StealingArtefactState.FAILURE)
+		{
+			client.clearHintArrow();
+		}
+		else
+		{
+			client.setHintArrow(stealingArtefactState.getWorldPoint());
+		}
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (event.getNpc().getId() == NpcID.CAPTAIN_KHALED_6972)
+		{
+			captainKhaled = event.getNpc();
+			if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+			{
+				client.setHintArrow(captainKhaled);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		if (captainKhaled != null && event.getNpc().getId() == captainKhaled.getId())
```

Comment:
This is not resolved. Per Adam's comment:

> This can just be a reference comparison with the stored npc.

```suggestion
		if (event.getNpc() == captainKhaled)
```